pkg/datarepair/checker/checker.go

```diff
@@ -146,3 +160,67 @@ func (c *checker) offlineNodes(ctx context.Context, nodeIDs storj.NodeIDList) (o
 	}
 	return offline, nil
 }
+
+// Find invalidNodes by checking the audit results that are place in statdb
+func (c *checker) invalidNodes(ctx context.Context, nodeIDs storj.NodeIDList) (invalidNodes []int32, err error) {
+	// filter if nodeIDs have invalid pieces from auditing results
+	findValidNodesReq := &sdbpb.FindValidNodesRequest{
+		NodeIds: nodeIDs,
+		MinStats: &pb.NodeStats{
+			AuditSuccessRatio: 0, // TODO: update when we have stats added to statdb
+			UptimeRatio:       0, // TODO: update when we have stats added to statdb
+			AuditCount:        0, // TODO: update when we have stats added to statdb
+		},
+	}
+
+	resp, err := c.statdb.FindValidNodes(ctx, findValidNodesReq)
+	if err != nil {
+		return nil, Error.New("error getting valid nodes from statdb %s", err)
+	}
+
+	var isValidNode bool
+	for i, node := range nodeIDs {
+		isValidNode = true
+
+		// Check if node is a valid node
+		for _, validNode := range resp.PassedIds {
+			if node == validNode {
+				break
+			}
+			isValidNode = false
+		}
+
+		// if node is not found in validNodes then the node is invalid
+		if !isValidNode {
+			invalidNodes = append(invalidNodes, int32(i))
+		}
+	}
+
+	return invalidNodes, nil
+}
+
+// combine the offline nodes with nodes marked invalid by statdb
+func combineOfflineWithInvalid(offlineNodes []int32, invalidNodes []int32) (missingPieces []int32) {
+	missingPieces = append(missingPieces, offlineNodes...)
+
+	var found bool
+	for _, invalidNode := range invalidNodes {
```

Comment:
you can get rid of the nested for loop here using the same process detailed in my other comment

Reply:
done